payment/src/main/java/org/killbill/billing/payment/core/sm/PaymentAutomatonDAOHelper.java

```diff
@@ -126,6 +126,16 @@ public void processPaymentInfoPlugin(final TransactionStatus transactionStatus,
         paymentStateContext.setPaymentTransactionModelDao(paymentDao.getPaymentTransaction(paymentStateContext.getPaymentTransactionModelDao().getId(), internalCallContext));
     }
 
+    public String getPaymentProviderPluginName() throws PaymentApiException {
+
+        final UUID paymentMethodId = paymentStateContext.getPaymentMethodId();
+        final PaymentMethodModelDao methodDao = paymentDao.getPaymentMethodIncludedDeleted(paymentMethodId, internalCallContext);
+        if (methodDao == null) {
+            throw new PaymentApiException(ErrorCode.PAYMENT_NO_SUCH_PAYMENT_METHOD, paymentMethodId);
+        }
+        return methodDao.getPluginName();
+    }
+
```

Comment:
Same is true here as above. It's not an additional DAO call. Just the retrieval of the plugin is moved to the calling method, so the name of the plugin doesn't get lost.